pylint/checkers/utils.py

```diff
@@ -802,3 +802,39 @@ def node_type(node):
     except astroid.InferenceError:
         return
     return types.pop() if types else None
+
+
+def is_singledispatch_function(node):
```

Comment:
I reimplemented explicit loop, because safe_infer used in decorated_with breaks when multiple values are inferred, and for singledispatch try/except with importing backport is very likely scenario which should be handled properly.